be/src/exprs/table_function/table_function_factory.cpp

```diff
@@ -92,14 +85,32 @@ const std::unordered_map<std::pair<std::string, bool>, std::function<TableFuncti
                 {{"explode_json_array_string", true}, VExplodeJsonArrayStringCreator},
                 {{"explode_bitmap", true},
                  TableFunctionCreator<vectorized::VExplodeBitmapTableFunction>()},
-                {{"explode", true}, VExplodeCreator},
-                {{"explode_outer", true}, VExplodeOuterCreator}}; // namespace doris
+                {{"explode", true}, TableFunctionCreator<vectorized::VExplodeTableFunction> {}}};
+
+Status TableFunctionFactory::get_fn(std::string fn_name, bool is_vectorized, ObjectPool* pool,
+                                    TableFunction** fn) {
+    bool is_outer = false;
+
+    auto match_suffix = [](const std::string& name, const std::string& suffix) -> bool {
+        return name.substr(name.length() - suffix.length()) == suffix;
+    };
+
+    auto remove_suffix = [](const std::string& name, const std::string& suffix) -> std::string {
+        return name.substr(0, name.length() - suffix.length());
+    };
+
+    if (match_suffix(fn_name, suffix_outer)) {
+        is_outer = true;
+        fn_name = remove_suffix(fn_name, suffix_outer);
+    }
 
-Status TableFunctionFactory::get_fn(const std::string& fn_name, bool is_vectorized,
```

Comment:
use a local variable, such as std::string search_fn_name for search the function.
Does it looks better than change argument of get_fn function?